### ISSUE TITLE
Error After Importing @polkadot/api

### ISSUE THREAD

**Issue by barmanabarmana** [polkadot-js/api#3938]
I was trying to integrate Polkadot-api to my website.but after importing there is errors in console.I think that this problem comes from package.json
This is what i was importing ` import { ApiRx } from '@polkadot/api';`

package.json code 
`"@polkadot/api": "^3.9.3",
        "@polkadot/api-derive": "^3.9.3",
        "@polkadot/rpc-core": "^3.9.3",
        "@polkadot/types": "^3.9.3",
        "@polkadot/types-known": "^3.9.3",
        "@polkadot/util": "^5.6.3",
        "@polkadot/dev": "^0.62.65",
        "@polkadot/extension-dapp": "^0.32.1",
        "@polkadot/keyring": "^3.7.1",
        "@polkadot/networks": "^7.3.1",
        "@polkadot/react-identicon": "^0.58.1",
        "@polkadot/rpc-provider": "^5.8.3",
        "@polkadot/ui-identicon": "^0.41.1",
        "@polkadot/ui-keyring": "^0.58.1",
        "@polkadot/ui-shared": "^0.85.3",
        "@polkadot/util-crypto": "^3.4.1",
        "@polkadot/x-fetch": "^7.3.1",
        "@polkadot/x-global": "^7.3.1",
        "@polkadot/x-textdecoder": "^7.3.1",
        "@polkadot/x-textencoder": "^7.3.1",
        "@polkadot/x-ws": "^7.3.1"`

![image](https://user-images.githubusercontent.com/87117376/132384104-9811e2d0-4233-419a-b966-3e970bc8cf65.png)

**Comment by jacogr** [comment]
You are mixing versions here - new RPC-provider, ancient API, util and util-crypto not matched. 

Ensure you have latest of all aligned, as it is set here you are in for a world of hurt. 

the error is caused by ancient API versions not able to use current util. (There has been major version bumps in-between, specifically since breakages were introduced)

**Comment by barmanabarmana** [comment]
@jacogr did all as you said.
this is my updated package.json.
And still alot of errors like was before 
![image](https://user-images.githubusercontent.com/87117376/132395687-28623140-6961-4fd7-960a-98157a060f5f.png)
![image](https://user-images.githubusercontent.com/87117376/132395707-0909569c-591e-4296-aaf9-82cf39181584.png)


``"@polkadot/api": "^5.8.3",
        "@polkadot/api-derive": "^3.11.1",
        "@polkadot/dev": "^0.62.66",
        "@polkadot/extension-dapp": "^0.32.1",
        "@polkadot/hw-ledger": "^6.11.1",
        "@polkadot/keyring": "^6.11.1",
        "@polkadot/networks": "^7.3.1",
        "@polkadot/react-identicon": "^0.58.1",
        "@polkadot/rpc-core": "^3.11.1",
        "@polkadot/rpc-provider": "^5.8.3",
        "@polkadot/types": "^3.11.1",
        "@polkadot/types-known": "^3.11.1",
        "@polkadot/ui-identicon": "^0.41.1",
        "@polkadot/ui-keyring": "^0.58.1",
        "@polkadot/ui-shared": "^0.85.3",
        "@polkadot/util": "^6.11.1",
        "@polkadot/util-crypto": "^6.11.1",
        "@polkadot/x-fetch": "^7.3.1",
        "@polkadot/x-global": "^7.3.1",
        "@polkadot/x-textdecoder": "^7.3.1",
        "@polkadot/x-textencoder": "^7.3.1", ``

**Comment by jacogr** [comment]
There is -

- no axios in Polkadot-js 
- no abortcontroller polyfill

Your rollup config cannot resolve. 

Nothing above is Polkadot-JS related.

**Comment by barmanabarmana** [comment]
@jacogr 
I tried to update to latest versions but i have same issue,millions of errors 
Can you show me please working rollup.config or example of a working project with rollup.config?so i can use it.
thank you

**Comment by jacogr** [comment]
I cannot give rollup support. I can point you to the rollup config which is used in the API bundling process to build browser bundles, YMMV since these is iffe specifically for libraries (However the plugins may yield something that pertains to your setup) -

- util bundles - https://github.com/polkadot-js/common/blob/master/rollup.config.mjs
- api bundles - https://github.com/polkadot-js/api/blob/master/rollup.config.mjs

Both these re-use (as does the extension and UI lib bundles) the functionality/base-config in https://github.com/polkadot-js/dev/blob/master/packages/dev/config/rollup.js

This is the process in action for the API https://github.com/polkadot-js/api/runs/3541319421?check_suite_focus=true#step:3:6171 and for the util/common repo https://github.com/polkadot-js/common/runs/3544672765?check_suite_focus=true#step:3:2485

As mentioned, your YMMV with your specific config (which is obviously different here, this is for bundling a library for browser use)

**Comment by barmanabarmana** [comment]
@jacogr 
I need to create bundle from module @polkadot/api for browser
Can you check my test repository that shows my error https://github.com/barmanabarmana/polkadot-test
Look at my bundle.js first line,there is few modules that are not included in the bundle:
```
_xTextdecoder, _xTextencoder, _xRandomvalues, _wasmCryptoAsmjs, _wasmCryptoWasm, buffer$3, crypto$2, _xFetch, _xWs

```
+In browser console i have this issue `_xTextdecoder is undefined`
Hope you can help me

**Comment by jacogr** [comment]
Like I said, I cannot provide rollup support.Tjis really has nothing to do with the API. 

In this case however a 1 min look showed exactly what you are missing -

1. Your plugin commonjs is outdated - bump to `^20.0.0` to enable all recent features
2. You plugin resolve is outdated - bump to `^13.0.4` to enable all recent features
3. ... have not checked, but assuming the other 2 are issues as well
4. index.ts change to `import { ApiPromise } from '@polkadot/api';` (actual TypeScript with ESM)

(after 1-3 ensure you run install after updating package.json)

Result, the build succeeds.

TL;DR Your rollup deps are way out of date

**Comment by barmanabarmana** [comment]
@jacogr 
Did all what you`ve recommended.
Only Buffer error left,tried to handle  like you did here.but unfortunately it doesnt work for me https://github.com/polkadot-js/common/blob/master/rollup.config.mjs#L44
![image](https://user-images.githubusercontent.com/87117376/132862651-a93f48ac-8213-4078-a970-f6121c4a631f.png)

**Comment by jacogr** [comment]
https://www.google.com/search?q=rollup+buffer+is+not+defined

**Comment by barmanabarmana** [comment]
who will have same problem which took me two weeks to handle
here is what might help you if all above didnt
npm i https://www.npmjs.com/package/rollup-plugin-node-builtins then add this to plugins in rollup.config.js
this worked for me
